### PR TITLE
add marker labels to connectome plot

### LINKED ISSUE #2733: plotting.view_markers marker_labels
I have been using `plotting.view_markers` for some time and really like the ability to define marker_color. I am wondering if it's possible to add marker_label as well?

### Benefits to the change
I think many users would find this helpful, especially when plotting many points at one time. In our use case, we are displaying depth electrode contacts for epilepsy patients and having each contact point labelled would be a benefit to our clinicians.

```python
import matplotlib.pyplot as plt
from nilearn import plotting

labels = #string label for each electrode contact
groups = #of implanted electrodes
n_members = #of contacts for each electrode

cmap = plt.get_cmap('rainbow')
colors=np.repeat(cmap(np.linspace(0, 1, len(groups))), n_members, axis=0)

html_view = plotting.view_markers(coordinates, marker_size=5.0, marker_color=colors, marker_labels=labels)
```

### PR DESCRIPTION
**Pull request by greydongilmore**
Closes #2733.

Changes proposed in this pull request: marker labels for the connectome plot.

-
-

### CODE REVIEW

File: nilearn/plotting/html_connectome.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def view_connectome(adjacency_matrix, node_coords, edge_threshold=None,
 
 
 def view_markers(marker_coords, marker_color=None, marker_size=5.,
-                 title=None, title_fontsize=25):
+                 marker_labels=None,title=None, title_fontsize=25):
```
Comment (NicolasGensollen) [inline nilearn/plotting/html_connectome.py:181]: I think there is a space missing before `title=`

**Review by NicolasGensollen — CHANGES_REQUESTED** [review]
Thanks a lot for adding this @greydongilmore ! :+1: 
I tried it locally and it seems to work correctly. 
I think we only need to add a test (in `nilearn/plotting/tests/test_html_connectome.py`) and a whatsnew entry (in `doc/whats_new.rst`).
For the test, unless you have a better idea, I think something like this should be enough:

```python
import json
import numpy as np
from nilearn.plotting import html_connectome

coords = np.arange(12).reshape((4, 3))
colors = ['r', 'g', 'black', 'white']
labels = ["red marker", "green marker", 
          "black marker", "white marker"]
html = html_connectome.view_markers(coords, 
                                    marker_size=5.0,
                                    marker_color=colors,
                                    marker_labels=labels)
labels_dict = {"marker_labels": labels}
assert json.dumps(labels_dict)[1:-1] in html.html
```

**Comment by greydongilmore** [comment]
@NicolasGensollen I've made the changes, it seems like some checks are now failing

**Comment by greydongilmore** [comment]
:man_facepalming: left out the json import statement.. everything seems okay now.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -108,6 +108,7 @@ Enhancements
   :func:`nilearn.reporting.get_clusters_table` have a new argument,
   "two_sided", which allows for two-sided thresholding, which is disabled by default.
 
+- :func:`nilearn.plotting.view_markers` now accept an optional argument `marker_labels` to provide labels to each marker.
```
Comment (NicolasGensollen) [inline doc/whats_new.rst:111]: Could you move this line into the Enhancements section of `0.7.2.dev` (beginning of the file)?

**Review by NicolasGensollen — CHANGES_REQUESTED** [review]
Thanks for adressing my reviews @greydongilmore !
This looks good to me (minor moving the line in `whats_new.rst`, see comment below).

**Review by bthirion — COMMENTED** [review]
LGTM besides the small glitch reported by @NicolasGensollen 
Thx !

File: doc/whats_new.rst (hunk 2 above)
Comment (greydongilmore) [reply doc/whats_new.rst]: @NicolasGensollen resolved

**Comment by greydongilmore** [comment]
@NicolasGensollen @bthirion resolved, my apologies

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @greydongilmore !

**Review by bthirion — APPROVED** [review]
Thx !

**Pull request merged**